upstairs/src/impacted_blocks.rs

```diff
@@ -119,83 +260,95 @@ impl ImpactedBlocks {
 ///    eid1 -> [0, 1, 2],
 ///  }
 pub fn extent_from_offset(
-    ddef: RegionDefinition,
+    ddef: &RegionDefinition,
     offset: Block,
     num_blocks: Block,
 ) -> ImpactedBlocks {
-    assert!(num_blocks.value > 0);
-    assert!(
-        (offset.value + num_blocks.value)
-            <= (ddef.extent_size().value * ddef.extent_count() as u64)
-    );
-    assert_eq!(offset.block_size_in_bytes() as u64, ddef.block_size());
+    let extent_size = ddef.extent_size().value;
 
-    let mut result = ImpactedBlocks::new(ddef);
-    let mut o: u64 = offset.value;
-    let mut blocks_left: u64 = num_blocks.value;
+    // I don't really think these assertions belong here, but we have a test
```

Comment:
Is this maybe need for the proptest stuff?  Like it might as part of the tests it does, send 
invalid offsets into `extent_from_offset()` and it needs that function to not panic?

Reply:
It is not needed for proptest stuff. We can do whatever we want with proptest stuff. I'm just not sure what we gain from randomly dying inside the ImpactedBlocks creation, so I opted for coercing input into something that makes sense for the region. But I can put it back to sudden-death mode if we prefer that.